File: uikit/src/main/java/io/agora/chat/uikit/interfaces/EaseMessageListener.java
package io.agora.chat.uikit.interfaces;

import java.util.List;

import io.agora.MessageListener;
import io.agora.chat.ChatMessage;

public abstract class EaseMessageListener implements MessageListener {
    @Override
    public abstract void onMessageReceived(List<ChatMessage> messages);

    @Override
    public void onCmdMessageReceived(List<ChatMessage> messages) {

    }

    @Override
    public void onMessageRead(List<ChatMessage> messages) {

    }

    @Override
    public void onMessageDelivered(List<ChatMessage> messages) {

    }

    @Override
    public abstract void onMessageRecalled(List<ChatMessage> messages);

    @Override
    public abstract void onMessageChanged(ChatMessage message, Object change);
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/interfaces/EaseMessageListener.kt
package io.agora.uikit.interfaces

import io.agora.uikit.common.ChatGroupReadAck
import io.agora.uikit.common.ChatMessage
import io.agora.uikit.common.ChatMessageListener
import io.agora.uikit.common.ChatMessagePinInfo
import io.agora.uikit.common.ChatMessagePinOperation
import io.agora.uikit.common.ChatMessageReactionChange

open class EaseMessageListener: ChatMessageListener {

    override fun onMessageReceived(messages: MutableList<ChatMessage>?) {}

    override fun onCmdMessageReceived(messages: MutableList<ChatMessage>?) {}

    override fun onMessageRead(messages: MutableList<ChatMessage>?) {}

    override fun onGroupMessageRead(groupReadAcks: MutableList<ChatGroupReadAck>?) {}

    override fun onReadAckForGroupMessageUpdated() {}

    override fun onMessageDelivered(messages: MutableList<ChatMessage>?) {}

    override fun onReactionChanged(messageReactionChangeList: MutableList<ChatMessageReactionChange>?) {}

    override fun onMessageContentChanged(
        messageModified: ChatMessage?,
        operatorId: String?,
        operationTime: Long
    ) {}

    override fun onMessagePinChanged(
        messageId: String?,
        conversationId: String?,
        pinOperation: ChatMessagePinOperation?,
        pinInfo: ChatMessagePinInfo?
    ) {
    }
}